packages/financial-templates-lib/src/proxy-transaction-handler/DSProxyManager.js

```diff
@@ -0,0 +1,148 @@
+const { getFromBlock } = require("@uma/common");
+
+class DSProxyManager {
+  /**
+   * @notice Constructs new Liquidator bot.
+   * @param {Object} logger Module used to send logs.
+   * @param {Object} web3 Web3 object to submit transactions and process on-chain info.
+   * @param {Object} gasEstimator Module used to estimate optimal gas price with which to send txns.
+   * @param {String} account Ethereum account from which to send txns.
+   * @param {String} dsProxyFactoryAddress address of the DSProxy factory to create new DSProxies and the like.
+   */
+  constructor({ logger, web3, gasEstimator, account, dsProxyFactoryAddress, dsProxyFactoryAbi, dsProxyAbi }) {
+    assert(web3.utils.isAddress(account), "Account needs to be a valid address");
+    assert(web3.utils.isAddress(dsProxyFactoryAddress), "dsProxyFactoryAddress needs to be a valid contract address");
+    this.logger = logger;
+    this.account = account;
+    this.web3 = web3;
+    this.gasEstimator = gasEstimator;
+    this.dsProxyFactoryAddress = dsProxyFactoryAddress;
+    this.dsProxyFactory = new web3.eth.Contract(dsProxyFactoryAbi, dsProxyFactoryAddress);
+    this.dsProxyAbi = dsProxyAbi;
+    this.dsProxy = null;
+    this.dsProxyAddress = null;
+
+    // Helper functions from web3.
+    this.BN = this.web3.utils.BN;
+    this.toBN = this.web3.utils.toBN;
+    this.toWei = this.web3.utils.toWei;
+    this.fromWei = this.web3.utils.fromWei;
+    this.utf8ToHex = this.web3.utils.utf8ToHex;
+    this.isAddress = this.web3.utils.isAddress;
+
+    // Multiplier applied to Truffle's estimated gas limit for a transaction to send.
+    this.GAS_LIMIT_BUFFER = 1.25;
```

Comment:
You should probably set this in a default config object like the other bots, but this is a nit and can be handled in a subsequent PR